back_ends/r/exec.R

```diff
@@ -28,24 +30,45 @@ if (file.exists(check_code_file_path)) {
   }
 }
 
-exec_jpg <- function(code) {
+exec_jpg <- function(code, i=0) {
   library(ggplot2)
 
   pdf(NULL) # Prevents Rplots.pdf from being created.
 
   eval(parse(text=code))
-  ggsave("/sandbox/image_output", dpi = 150, device = "jpeg")
-}
-
-code <- readChar(code_file_path, file.info(code_file_path)$size)
-
-if (file.exists(test_code_file_path)) {
-  test_code <- readChar(test_code_file_path, file.info(test_code_file_path)$size)
-  code <- paste(code, test_code, sep="\n\n")
+  if (i == 0) {
+    ggsave("/sandbox/image_output", dpi = 150, device = "jpeg")
+  }
+  else {
+    ggsave(paste("/sandbox/test_image_output_", i, sep=""), dpi = 150, device = "jpeg")
+  }
 }
 
 if (output_type == "txt") {
-  suppressMessages(suppressWarnings(suppressPackageStartupMessages(eval(parse(text=code)))))
+  system(paste("Rscript", code_file_path)) # run code using Rscript
 } else {
+  code <- readChar(code_file_path, file.info(code_file_path)$size)
   suppressMessages(suppressWarnings(suppressPackageStartupMessages(exec_jpg(code))))
```

Comment:
I found out that all the suppressMessages and Warnings wrappers around ```exec_jpg``` are actually necessary to avoid getting notifications that say ```Saving 7 x 7 in image
`geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'``` so I left them in.